test/optim/test_optim.py

```diff
@@ -800,53 +802,29 @@ def test_multi_tensor_optimizers(self):
             (optim.Adagrad, dict(weight_decay=1)),
         ]
-        self._test_derived_optimizers(optimizer_pairs_with_flags, "foreach")
+
+    def test_multi_tensor_optimizers(self):
+        self._test_derived_optimizers(self._multi_tensor_optimizer_configs, "foreach")
 
     @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
     def test_multi_tensor_optimizers_with_varying_tensors(self):
-        optimizer_pairs_with_flags = [
-            (optim.Adam, dict(weight_decay=1.0, amsgrad=True, fused=False)),
-            (optim.Adam, dict(weight_decay=1.0, amsgrad=False, fused=False)),
-            (optim.Adam, dict(weight_decay=0.0, amsgrad=True, fused=False)),
-            (optim.Adam, dict(weight_decay=0.0, amsgrad=False, fused=False)),
-            (optim.AdamW, dict(weight_decay=1.0, amsgrad=True)),
-            (optim.AdamW, dict(weight_decay=1.0, amsgrad=False)),
-            (optim.AdamW, dict(weight_decay=0.0, amsgrad=True)),
-            (optim.AdamW, dict(weight_decay=0.0, amsgrad=False)),
-            (optim.NAdam, dict(weight_decay=0.0, momentum_decay=6e-3)),
-            (optim.NAdam, dict(weight_decay=1.0, momentum_decay=6e-3)),
-            (optim.NAdam, dict(weight_decay=0.0, momentum_decay=4e-3)),
-            (optim.NAdam, dict(weight_decay=0.01, momentum_decay=4e-3)),
-            (
-                optim.SGD,
-                dict(lr=0.2, momentum=1, dampening=0, weight_decay=1, nesterov=True),
-            ),
-            (
-                optim.SGD,
-                dict(lr=0.2, momentum=1, dampening=0.5, weight_decay=1, nesterov=False),
-            ),
-            (optim.RAdam, dict(weight_decay=0, eps=1e-6)),
-            (optim.RAdam, dict(weight_decay=0)),
-            (optim.RAdam, dict(weight_decay=1, eps=1e-6)),
-            (optim.RAdam, dict(weight_decay=1)),
-            (optim.RMSprop, dict(weight_decay=1, momentum=1, centered=True)),
-            (optim.RMSprop, dict(weight_decay=1, momentum=0, centered=True)),
-            (optim.RMSprop, dict(weight_decay=1, momentum=1, centered=False)),
-            (optim.RMSprop, dict(weight_decay=0, momentum=1, centered=False)),
-            (optim.Rprop, dict(lr=1e-2, etas=(0.5, 1.2), step_sizes=(1e-6, 50))),
-            (optim.ASGD, dict(weight_decay=0)),
-            (optim.ASGD, dict(weight_decay=1)),
-            (optim.Adamax, dict(weight_decay=0)),
-            (optim.Adamax, dict(weight_decay=1)),
-            (optim.Adadelta, dict(weight_decay=0)),
-            (optim.Adadelta, dict(weight_decay=1)),
-            (optim.Adagrad, dict(weight_decay=0)),
-            (optim.Adagrad, dict(weight_decay=1)),
-        ]
-        self._test_derived_optimizers_varying_tensors(optimizer_pairs_with_flags, "foreach")
+        self._test_derived_optimizers_varying_tensors(self._multi_tensor_optimizer_configs, "foreach")
+
+    @unittest.skipIf(not torch.cuda.is_available(), "Requires a GPU")
+    @largeTensorTest("72GB", "cuda")
```

Comment:
can you guys make sure that you run this test, as it won't be run in CI?
Also, the original repro required less memory?

Reply:
We run tests on A100 on a daily basis so it's likely to be tested.

I found this way faster than the repro script. Not sure if it's slimmer than this.

Reply:
Ok, but can I ask you to promptly share failures of your internal tests? In the past, we've frequently been in the situation where we discover the bug on the main all too late, and you guys saw it and didn't report. cc @ptrblck, can we device a more formal process of sharing the test failures that you are seeing?